### DISCUSSION TITLE
Is it possible to have fumadocs running while using src directory?

### DISCUSSION THREAD

**Discussion by kyrylolvov** [fuma-nama/fumadocs#451] [Q&A] [answered]
I am trying to make `pnpm create fumadocs-app` run when moving app and content inside src, changing paths, but when visiting /docs it returns 404.

What should be specified in `loader()` for it to run properly?

**Comment by fuma-nama** [comment]
Modify [rootDir](https://fumadocs.vercel.app/docs/headless/source-api#root-directory)

**Reply by kyrylolvov** [reply]
@fuma-nama I have tried setting `rootDir` to "", "src", "src/content". But they all just return an empty array.

Thats the project strcture:
<img width="242" alt="Screenshot 2024-06-12 at 10 19 50 AM" src="https://github.com/fuma-nama/fumadocs/assets/70720713/bf8461fa-fc56-4097-9474-75ac1121cad5">

**Reply by fuma-nama** [reply, marked as answer]
Oops I misunderstood the question, for this, you have to modify your next.config file and set `rootContentPath` instead. Because by default we only scan the files inside the content directory.

**Reply by kyrylolvov** [reply]
@fuma-nama Thank you so much! Sorry if the question was misleading.

**Comment by kyrylolvov** [comment]
Will leave this here, if anyone wants to move `content` inside `src`, you need to update createMDX call in config.

`const withMDX = createMDX({ rootContentPath: "./src/content" });`

**Reply by fuma-nama** [reply]
I've just documented this, see https://fumadocs.vercel.app/docs/mdx/configuration. Hope it helps!

**Reply by CXL-edu** [reply]
Excuse me, has it been updated? I tried to find that there is no rootContentPath parameter?

> I've just documented this, see https://fumadocs.vercel.app/docs/mdx/configuration. Hope it helps!

**Reply by fuma-nama** [reply]
This is long time ago before Fumadocs MDX v10 (which is a big redesign), now it's way easier, you can simply use the `dir` option of `defineDocs` in `source.config.ts`

**Reply by CXL-edu** [reply]
> This is long time ago before Fumadocs MDX v10 (which is a big redesign), now it's way easier, you can simply use the `dir` option of `defineDocs` in `source.config.ts`

I tried to replace the `dir` option of `defineDocs` in `source.config.ts` file from `content/docs` to `src/content/docs`, I encountered a `404 - Page Not Found` error.

**Reply by fuma-nama** [reply]
you can try to console log the `docs` array to see if it's correct, like:

```ts
// lib/source.ts
import { createMDXSource } from 'fumadocs-mdx';
import { loader } from 'fumadocs-core/source';
import { docs, meta } from '@/.source';

console.log(docs, meta)

export const source = loader({
  baseUrl: '/docs',
  source: createMDXSource(docs, meta),
});
```

**Reply by CXL-edu** [reply]
I found the issue - it's caused by the command below. The `.source` folder is in the project root directory, while the @ command uses `src` as the root directory.

When I changed `@/.source` to `../../.source`, the page could find it, but there were layout errors. Perhaps there are path issues within the `.source` folder.

```typescripts
// lib/source.ts
...
`import { docs, meta } from '@/.source';`
...
```
Do you have any suggestions?

**Reply by fuma-nama** [reply]
Hmm what layout errors?
